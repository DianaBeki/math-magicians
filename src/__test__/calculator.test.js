import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../components/calculator';

test('test calculator components work correctly', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchSnapshot();
});
