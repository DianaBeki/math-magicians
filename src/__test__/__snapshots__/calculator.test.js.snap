// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="calc-container"
    style="background-color: lightgrey; width: 50%; display: flex; flex-direction: column; margin: 4.5rem auto; position: relative; padding: 3rem 0px;"
  >
    <div
      style="margin-top: -1.8rem; margin-left: 2rem; position: absolute; color: orange; font-weight: 600;"
    >
      <h2>
        Math Magician
      </h2>
      <span>
        Lets do some math!
      </span>
    </div>
    <div
      class="calculator"
    >
      <div
        class="display"
      >
        0
      </div>
      <div
        class="button-container"
      >
        <button
          class="button"
          type="button"
        >
          AC
        </button>
        <button
          class="button"
          type="button"
        >
          +/-
        </button>
        <button
          class="button"
          type="button"
        >
          %
        </button>
        <button
          class="orange"
          type="button"
        >
          รท
        </button>
        <button
          class="button"
          type="button"
        >
          7
        </button>
        <button
          class="button"
          type="button"
        >
          8
        </button>
        <button
          class="button"
          type="button"
        >
          9
        </button>
        <button
          class="orange"
          type="button"
        >
          x
        </button>
        <button
          class="button"
          type="button"
        >
          4
        </button>
        <button
          class="button"
          type="button"
        >
          5
        </button>
        <button
          class="button"
          type="button"
        >
          6
        </button>
        <button
          class="orange"
          type="button"
        >
          -
        </button>
        <button
          class="button"
          type="button"
        >
          1
        </button>
        <button
          class="button"
          type="button"
        >
          2
        </button>
        <button
          class="button"
          type="button"
        >
          3
        </button>
        <button
          class="orange"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="btn-row"
      >
        <div
          class="button0"
          colspan="2"
        >
          <button
            class="button"
            type="button"
          >
            0
          </button>
        </div>
        <button
          class="button"
          type="button"
        >
          .
        </button>
        <button
          class="orange bg"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test calculator components work correctly 1`] = `
<DocumentFragment>
  <div
    class="calc-container"
    style="background-color: lightgrey; width: 50%; display: flex; flex-direction: column; margin: 4.5rem auto; position: relative; padding: 3rem 0px;"
  >
    <div
      style="margin-top: -1.8rem; margin-left: 2rem; position: absolute; color: orange; font-weight: 600;"
    >
      <h2>
        Math Magician
      </h2>
      <span>
        Lets do some math!
      </span>
    </div>
    <div
      class="calculator"
    >
      <div
        class="display"
      >
        0
      </div>
      <div
        class="button-container"
      >
        <button
          class="button"
          type="button"
        >
          AC
        </button>
        <button
          class="button"
          type="button"
        >
          +/-
        </button>
        <button
          class="button"
          type="button"
        >
          %
        </button>
        <button
          class="orange"
          type="button"
        >
          รท
        </button>
        <button
          class="button"
          type="button"
        >
          7
        </button>
        <button
          class="button"
          type="button"
        >
          8
        </button>
        <button
          class="button"
          type="button"
        >
          9
        </button>
        <button
          class="orange"
          type="button"
        >
          x
        </button>
        <button
          class="button"
          type="button"
        >
          4
        </button>
        <button
          class="button"
          type="button"
        >
          5
        </button>
        <button
          class="button"
          type="button"
        >
          6
        </button>
        <button
          class="orange"
          type="button"
        >
          -
        </button>
        <button
          class="button"
          type="button"
        >
          1
        </button>
        <button
          class="button"
          type="button"
        >
          2
        </button>
        <button
          class="button"
          type="button"
        >
          3
        </button>
        <button
          class="orange"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="btn-row"
      >
        <div
          class="button0"
          colspan="2"
        >
          <button
            class="button"
            type="button"
          >
            0
          </button>
        </div>
        <button
          class="button"
          type="button"
        >
          .
        </button>
        <button
          class="orange bg"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
